ml_engine/xgboost_model.py

```
@@ -34,56 +58,279 @@ class PlacementPredictor:
+        if active_scaler is not None:
+            feat_arr = active_scaler.transform(feat_arr)
+
+        # Compute prediction and probability
+        proba = float(self.model.predict_proba(feat_arr)[0, 1])
+        is_placed = bool(proba >= 0.5)
+
+        # Feature importances breakdown
+        importances = getattr(self.model, "feature_importances_", None)
+        if importances is not None:
+            feature_imp = {
+                name: float(imp) for name, imp in zip(FEATURE_NAMES, importances)
```

Ruff (0.16.3)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)